float64 diameter
time timestamp
